### PR TITLE
[MDS] Add dashboard assistant tests with multiple data source enabled

### PR DESCRIPTION
**Pull request by yubonluo**
### Description

Add dashboard assistant tests with multiple data source enabled

### Issues Resolved

[List any issues this PR will resolve]

### Check List

- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/support/index.js [hunk 1]
```diff
@@ -75,3 +78,25 @@ if (Cypress.env('DASHBOARDS_ASSISTANT_ENABLED')) {
...
+ */
+if (
+  Cypress.env('DASHBOARDS_ASSISTANT_ENABLED') &&
+  Cypress.env('DATASOURCE_MANAGEMENT_ENABLED')
+) {
+  const originalBackendEndpoint = currentBackendEndpoint.get();
+  before(() => {
+    currentBackendEndpoint.set(currentBackendEndpoint.REMOTE_NO_AUTH);
+    cy.addAssistantRequiredSettings();
+    cy.readOrRegisterRootAgent();
```
Comment (wanglam) [inline cypress/support/index.js:94]: Should set `currentBackendEndpoint` to original after readOrRegisterRootAgent, or this backend endpoint will be overwrite in all requests.

File: cypress/support/index.js [hunk 2]
```diff
@@ -75,3 +78,25 @@ if (Cypress.env('DASHBOARDS_ASSISTANT_ENABLED')) {
...
+    cy.startDummyServer();
+  });
+  after(() => {
+    currentBackendEndpoint.set(originalBackendEndpoint, false);
```
Comment (wanglam) [inline cypress/support/index.js:98]: Should call `currentBackendEndpoint.set(currentBackendEndpoint.REMOTE_NO_AUTH);` here.

File: cypress/support/index.js [hunk 3]
```diff
@@ -75,3 +78,25 @@ if (Cypress.env('DASHBOARDS_ASSISTANT_ENABLED')) {
...
+    cy.cleanRootAgent();
```
Comment (wanglam) [inline cypress/support/index.js:99]: Should reset to original here.

File: cypress/integration/plugins/dashboards-assistant/mds_conversation_save_to_notebook_spec.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { BASE_PATH } from '../../../utils/constants';
+import { setStorageItem } from '../../../utils/plugins/dashboards-assistant/helpers';
+
+const QUESTION = 'What are the indices in my cluster?';
+const FINAL_ANSWER = 'The indices in';
+
+if (
+  Cypress.env('DASHBOARDS_ASSISTANT_ENABLED') &&
```
Comment (raintygao) [inline cypress/integration/plugins/dashboards-assistant/mds_conversation_save_to_notebook_spec.js:12]: Save to notebook feature will hide when MDS enabled. We only need to test this function when MDS disabled for now.

**Review by raintygao — APPROVED** [review]

File: cypress/utils/plugins/dashboards-assistant/commands.js [hunk 5]
```diff
@@ -197,6 +197,16 @@ Cypress.Commands.add('sendAssistantMessage', (body) =>
   apiRequest(`${BASE_PATH}${ASSISTANT_API.SEND_MESSAGE}`, 'POST', body)
 );
 
+Cypress.Commands.add(
+  'sendAssistantMessageByDataSourceId',
+  (body, dataSourceId) =>
+    apiRequest(
+      `${BASE_PATH}${ASSISTANT_API.SEND_MESSAGE}?dataSourceId=${dataSourceId}`,
```
Comment (raintygao) [inline cypress/utils/plugins/dashboards-assistant/commands.js:204]: Do we need to add dataSourceId query to other commands?

**Pull request closed without merging**